### PR TITLE
Support for Custom Job resources

### PR DESCRIPTION
**Pull request by johnugeorge**
This PR adds support for TF v1beta2 and PyTorch v1beta2 job. Pytorch example is added. 
/assign @richardsliu

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/512)

### CODE REVIEW

File: pkg/common/v1alpha2/common.go [hunk 1]
```diff
@@ -29,3 +31,24 @@ func GetManagerAddr() string {
...
+func GetSupportedJobList() []schema.GroupVersionKind {
+	supportedJobList := []schema.GroupVersionKind{
+		schema.GroupVersionKind{
+			Group:   "batch",
+			Version: "v1",
+			Kind:    "Job",
+		},
+		schema.GroupVersionKind{
+			Group:   "kubeflow.org",
+			Version: "v1beta2",
```
[the remark this answers is not in the document]
Comment (johnugeorge) [reply pkg/common/v1alpha2/common.go]: I think, it doesn't matter much to Katib. We can also change to v1 version just before 0.6 release

File: pkg/controller/v1alpha2/experiment/util/webhook_util.go [hunk 2]
```diff
@@ -22,11 +22,13 @@ import (
...
 	logger "log"
 
-	commonv1alpha2 "github.com/kubeflow/katib/pkg/api/operators/apis/common/v1alpha2"
-	ep_v1alpha2 "github.com/kubeflow/katib/pkg/api/operators/apis/experiment/v1alpha2"
 	batchv1beta "k8s.io/api/batch/v1beta1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	k8syaml "k8s.io/apimachinery/pkg/util/yaml"
+
+	commonapi "github.com/kubeflow/katib/pkg/api/operators/apis/common/v1alpha2"
+	ep_v1alpha2 "github.com/kubeflow/katib/pkg/api/operators/apis/experiment/v1alpha2"
```
Comment (gaocegege) [inline pkg/controller/v1alpha2/experiment/util/webhook_util.go]: How about experimentsv1alpha2 and commonapiv1alpha2

Comment (johnugeorge) [reply pkg/controller/v1alpha2/experiment/util/webhook_util.go]: done

**Comment by gaocegege** [comment]
@johnugeorge Please rebase

**Comment by gaocegege** [comment]
It should be blocked by #523

**Pull request merged**